hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -576,7 +575,7 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 
 	// 2. - Reconcile towards expected state of the world.
 	targetConfigVersionHash := hashStruct(config + targetVersion)
-	compressedConfig, err := compress([]byte(config))
+	compressedConfig, err := supportutil.Compress([]byte(config))
```

Comment:
can you clarify why can't con compressAndEncode everywhere?

Reply:
The reason why we can't do CompressAndEncode everywhere is most likely due to a chicken-and-egg issue that happens whenever an upgrade occurs. What I think is happening is:

1. The Hypershift Operator is upgraded.
2. The upgraded nodepool controller reconciles, which compresses and encodes the payload.
3. The Hypershift Operator has not yet rolled out the new Ignition server pods, so the old ones are still active.
4. The old Ignition server pods don't know what to do with the compressed and encoded content, hence the gzip errors we were seeing.

For evidence, look at the image pullspec for the [Hypershift operator pod spec](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_hypershift/1671/pull-ci-openshift-hypershift-main-e2e-aws/1573397814489124864/artifacts/e2e-aws/run-e2e/artifacts/TestUpgradeControlPlane_PreTeardownClusterDump/namespaces/hypershift/core/pods/operator-5cbfff58c-s84mv.yaml). It points to the image that was built as part of the CI run (see: [image build logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_hypershift/1671/pull-ci-openshift-hypershift-main-e2e-aws/1573397814489124864/artifacts/build-logs/hypershift-operator.log)). Whereas, if you look at the image pullspec for the [Ignition server pod spec](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_hypershift/1671/pull-ci-openshift-hypershift-main-e2e-aws/1573397814489124864/artifacts/e2e-aws/run-e2e/artifacts/TestUpgradeControlPlane_PreTeardownClusterDump/namespaces/e2e-clusters-sqs2p-example-5g6mj/core/pods/ignition-server-87f44b8ff-2f88l.yaml), the image is from the nightly OpenShift nightly build we're upgrading from.